service-app/internal/api/client_test.go

```diff
@@ -0,0 +1,176 @@
+package api
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/ministryofjustice/opg-scanning/config"
+	"github.com/ministryofjustice/opg-scanning/internal/httpclient"
+	"github.com/ministryofjustice/opg-scanning/internal/logger"
+	"github.com/ministryofjustice/opg-scanning/internal/types"
+)
+
+type requestCaseStub struct {
+	name        string
+	xmlPayload  string
+	expectedReq *types.ScannedCaseRequest
+	expectedErr error
```

Comment:
I wonder if `expectedErr` should just be a boolean, since we don't compare the value of errors?

Reply:
Agreed, we can simplify